### PR TITLE
Removed use of 'time'

### LINKED ISSUE #130: Don't use both `time` and `chrono`.
You're using both the `time` library, and the `chrono` library. Don't use both, please.

### PR DESCRIPTION
**Pull request by jD91mZM2**
This fixes #130.

I do wonder however... Is this really the best solution? I feel like chrono is a huge library taking care of leap seconds and all that good stuff, but maybe I should be using the built in std::time instead?

You decide what you think is best!

**Warning**: It compiles, but functionality untested. And even if I did test it I wouldn't know what signs to look for that it worked.

### CODE REVIEW

**Comment by Xaeroxe** [comment]
@legolord208 Chrono has much better support for manipulation of time and duration values.  I'm in favor of its continued usage.

**Comment by SpaceManiac** [comment]
+1 for chrono on `creation_date`, that's a good unification of public API (since everything else exposes `chrono` types).

Now, it looks like this code was wrong before, but if we're already here it should probably be fixed - Timer should probably be using monotonic `std::time::Instant` rather than a timestamp, to guard against the clock going backwards. This code was originally written before it was stable.

Ratelimit tracking using UTC is correct. Long-term I want it to use `Instant` too and never compare our clock to the server's clock directly, but that's rather involved and out of scope here. I don't expect it in this PR.

**Comment by jD91mZM2** [comment]
Will work on the features later. Happy Rust 1.19 by the way :smile:

**Comment by jD91mZM2** [comment]
There you go! You have no idea how much I suddenly dislike the auto-formatting feature of the text editor I use haha :stuck_out_tongue:

File: src/lib.rs [hunk 1]
```diff
@@ -1176,44 +1176,47 @@ fn resolve_invite(invite: &str) -> &str {
...
 		}
 	}
 
 	fn sleep_until_tick(&mut self) {
-		let difference = self.next_tick_at - time::get_time();
-		if difference > time::Duration::zero() {
-			sleep_ms(difference.num_milliseconds() as u64)
+		let diff = self.next_tick_at - time::Instant::now();
+		if diff > time::Duration::new(0, 0) {
+			sleep_ms((diff.as_secs() * 1000 + (diff.subsec_nanos() / 1000000) as u64) as u64)
```
Comment (SpaceManiac) [inline src/lib.rs]: Just calling `std::thread::sleep(diff);` should work here?

Comment (jD91mZM2) [reply src/lib.rs]: Hmm lol that's true. Will take care of that tomorrow!

**Comment by SpaceManiac** [comment]
Looks good as a first pass. Couple of changes here I want to test first so I'll post back after that.

**Comment by jD91mZM2** [comment]
Check out #133 also. It adds Travis CI: Automated testing!

**Comment by jD91mZM2** [comment]
Hmm it panicked? Type bounds? Because I just translated the existing code. I didn't edit any logic...

**Comment by SpaceManiac** [comment]
`std::time::Duration` is unsigned compared to `time::Duration`.

**Comment by jD91mZM2** [comment]
o ok

**Pull request closed without merging**